test/restart.jl

```diff
@@ -0,0 +1,103 @@
+    # Generate a simulation with some complexity of 
+    # config arguments. Note that this does not test
+    # the configuration itself (there are many such
+    # combinations of config args which will generate
+    # different simulation objects).
+    
+    # Modify the timestep to 1-second increments. 
+    # Save simulation state at each timestep, 
+    # and generate a restart file at 2secs simulation time.
+    # Verify objects are identical (i.e. restart results 
+    # in the same simulation state as if one were to advance
+    # the timestepper uninterrupted.)
+    
+    # We don't test for the `parameter/ cache` since
+    # the `precomputed_quantities` function needs to be 
+    # called to ensure their identical states. This is tested
+    # elsewhere. 
+    
+    mkpath("./test_restart/")
+
+    ### Boilerplate default integrator objects
+    config = CA.AtmosConfig(
+        Dict(
+            "moist" => "nonequil",
+            "precip_model" => "0M",
+            "config" => "column",
+            "output_default_diagnostics" => false,
+            "dt" => "1secs",
+            "t_end" => "3secs",
+            "dt_save_state_to_disk" => "2secs",
+            "dt_save_state_to_sol" => "1secs", # Save every timestep
+            "output_dir" => "./test_restart/"
+        ),
+    )
+
+    @info config.parsed_args["output_dir"]
+
+    simulation = CA.get_simulation(config);
+
+    SciMLBase.solve!(simulation.integrator);
+
+    restart_dir = "./test_restart/output_active/"
+    restart₀ = joinpath(restart_dir,"day0.0.hdf5")
+    config₀ = CA.AtmosConfig(
+        Dict(
+            "moist" => "nonequil",
+            "precip_model" => "0M",
+            "config" => "column",
+            "output_default_diagnostics" => false,
+            "dt" => "1secs",
+            "t_end" => "3secs",
+            "dt_save_state_to_disk" => "2secs",
+            "dt_save_state_to_sol" => "1secs", # Save every timestep
+            "restart_file" => restart₀,
+        ),
+    )
+    simulation_test₀ = CA.get_simulation(config₀);
+    SciMLBase.step!(simulation_test₀.integrator);
+    SciMLBase.step!(simulation_test₀.integrator);
+    SciMLBase.step!(simulation_test₀.integrator);
+
+    restart₁ = joinpath(restart_dir,"day0.2.hdf5")
+    config₁ = CA.AtmosConfig(
+        Dict(
+            "moist" => "nonequil",
+            "precip_model" => "0M",
+            "config" => "column",
+            "output_default_diagnostics" => false,
+            "dt" => "1secs",
+            "t_end" => "3secs",
+            "dt_save_state_to_disk" => "2secs",
+            "dt_save_state_to_sol" => "1secs", # Save every timestep
+            "restart_file" => restart₁,
+        ),
+    )
+    simulation_test₁ = CA.get_simulation(config₁);
+    SciMLBase.step!(simulation_test₁.integrator);
+    
+    Y = simulation.integrator.u
+    Y₀ = simulation_test₀.integrator.u
+    Y₁ = simulation_test₁.integrator.u
+
+    rm("./test_restart", recursive=true)
```

Comment:
If the test fails, this will not be executed and some stray files will be left on the disk.

You should use temporary directories instead.